### DISCUSSION TITLE
ChromVAR analysis with PWM score thresholds

### DISCUSSION THREAD

**Discussion by Y-ogativity** [GreenleafLab/ArchR#1603] [Questions / Documentation] [answered]
Hi, 
Thank you for developing a great package!
I want to know how to conduct ChromVAR analysis with the custom PWMs (e.g., TF motifs identified by de novo motif discovery) with log odds (i.e., PWM score) thresholds instead of pval. 

One way to do so is to use the addDeviationsMatrix function with custom peakAnnotation matches. 
I looked into motifmatchr::matchmotifs(), but I did not find anything to make the matches data with log odds score.

I would be glad to get suggestions and comments from you guys. Thank you very much!

Yohei

**Comment by rcorces** [comment, marked as answer]
Sorry I'm not sure I understand your question. It seems to me like the answer you proposed is the solution. Are you saying that this approach does not work for you?

> One way to do so is to use the addDeviationsMatrix function with custom peakAnnotation matches.

https://www.archrproject.com/bookdown/archr-and-custom-deviations.html

**Reply by Y-ogativity** [reply]
Exactly. Sorry for your confusion. 

My goal is to use ChromVAR with de novo motifs (PWMs) with log odds thresholds for each motif. I would like to know how I can make the custom peakAnnotation matches according to the custom log odds score. I have predetermined PWM thresholds for each PWM, while motifmatcher only accepts for p-value thresholds instead of log odds.

**Reply by rcorces** [reply]
This is outside of the support that I can provide. ArchR provides a way for you to do deviations based on custom bed-style input. So your job is to figure out how to get the set of regions you are interested in and use that set of regions as input for a custom deviations analysis.

**Reply by Y-ogativity** [reply]
I see your point. I used another R package to prepare bed-style input and add custom peak annotation in ArchR. [The page in the ArchR tutorial](https://www.archrproject.com/bookdown/custom-enrichment.html) is beneficial. Thank you very much!
